apps/sms/services/js/service_mixin.js

```diff
@@ -0,0 +1,56 @@
+/*global bridge */
+
+'use strict';
+
+if (!('bridge' in self)) {
+  importScripts('/lib/bridge.js');
+}
+
+(function(exports) {
+
+  /**
+   * This mixin allows to expose easily a service from an object.
+   *
+   * This needs the following properties to work properly:
+   *
+   * - SERVICE_NAME holds the service name.
+   * - METHODS holds the methods that will be exposed by the service.
+   * - STREAMS holds the methods that handle streams exposed by the service.
+   *
+   * EVENTS, METHODS and STREAMS are all optional.
+   */
+  var serviceMixin = {
+    initService() {
```

Comment:
Mmm, why not pass ```name```, ```methods```, ```streams``` as parameters to this method? I'd rather keep this constants as private like:
```
const SERVICE_NAME = 'service';
const METHODS = ['method'];

var Service = {
  init() {
    this.initService(SERVICE_NAME, METHODS);
  }
}
```